### PR TITLE
cli: nicer address selection

### PR DESCRIPTION
**Pull request by hackaugusto**
addresses

### CODE REVIEW

**Comment by LefterisJP** [comment]
Reviewing this.

File: raiden/ui/cli.py [hunk 1]
```diff
@@ -803,35 +808,32 @@ def prompt_account(address_hex, keystore_path, password_file):
...
+    if not address_hex:
         formatted_addresses = [
-            '[{:3d}] - {}'.format(idx, to_checksum_address(addr))
-            for idx, addr in enumerate(addresses)
+            to_checksum_address(addr)
+            for addr in accmgr.accounts.keys()
         ]
 
-        should_prompt = True
+        addresses = inquirer.List(
```
Comment (LefterisJP) [inline raiden/ui/cli.py]: Can you add an echo above this? To serve the purpose of instructions of how the user selects the address now.

`print('Use the arrow keys to select the address to be used by Raiden')`

File: raiden/ui/cli.py [hunk 2]
```diff
@@ -803,35 +808,32 @@ def prompt_account(address_hex, keystore_path, password_file):
...
-                address_hex))
-            sys.exit(1)
-
-        addresses = list(accmgr.accounts.keys())
+    if not address_hex:
         formatted_addresses = [
-            '[{:3d}] - {}'.format(idx, to_checksum_address(addr))
-            for idx, addr in enumerate(addresses)
+            to_checksum_address(addr)
+            for addr in accmgr.accounts.keys()
```
Comment (ulope) [inline raiden/ui/cli.py]: While you're at it, could you add a `sorted()` call here? Having the addresses in random order has always been annoying.

**Comment by ulope** [comment]
The cli tests also need to be updated. (BTW what does inquirer do when running non-interactively?)

**Comment by hackaugusto** [comment]
> BTW what does inquirer do when running non-interactively?

It blocks. But the user is supposed to provide the address through the command line in order to Raiden to not show the menu:

https://github.com/raiden-network/raiden/blob/f8efb479f53f3d82d36ec6a4fc59641953b9ccc1/raiden/ui/cli.py#L811

**Comment by LefterisJP** [comment]
Yeah the CLI tests need to be updated to provide address via the CLI and then bypass the expectations of address selection.

**Comment by hackaugusto** [comment]
The build is green and the tests passed, which tests are you guys referring to?

https://travis-ci.org/raiden-network/raiden/jobs/428545060#L1161
https://travis-ci.org/raiden-network/raiden/jobs/428545060#L1183
https://travis-ci.org/raiden-network/raiden/jobs/428545060#L1190

**Comment by LefterisJP** [comment]
The build is green because you did not run with `[ci integration]` ;)

I am referring to [this](https://github.com/raiden-network/raiden/blob/51ffd61082dcdb5ada484c867fb572bcf8d95329/raiden/tests/integration/cli/test_cli.py#L46) part of the CLI integration tests.

File: docs/changelog.rst [hunk 3]
```diff
@@ -11,6 +11,7 @@ Changelog
 * :bug:`2427` Fix a bug crashing the client when an unlock event for our address is seen on the chain
 * :bug:`2431` Do not crash on recoverable errors during settlement
 * :feature:`2460` Pinned depedencies versions, builds are now reproducible and build artifacts won't break because of downstream dependencies.
+* :feature:`2484` Nicier visual address selection for the CLI.
```
Comment (rakanalh) [inline docs/changelog.rst]: s/Nicier/Nicer

**Comment by LefterisJP** [comment]
Fixes #2484

**Pull request closed without merging**